### PR TITLE
[#2528] Add categories to languages

### PR DESCRIPTION
**Pull request by arbron**
Groups languages into categories based on how they are defined in the SRD.

Language Selector
Language Advancement

This change does modify the `languages` object in config, but it shouldn't be a breaking change for most cases because adding languages in the old fashion (`CONFIG.DND5E.languages.test = "test";`) will still work correctly (those languages will just appear at the top level.

Retrieving languages using `Trait.keyLabel` will also continue working. The only case that will lead to unexpected results is if code was retrieving languages directly (`console.log(CONFIG.DND5E.languages.common`), but that will just return empty.

### CODE REVIEW

File: module/config.mjs [hunk 1]
```diff
@@ -1683,38 +1683,62 @@ DND5E.conditionTypes = {
...
+  },
+  exotic: {
+    label: "DND5E.LanguagesExotic",
+    children: {
+      aarakocra: "DND5E.LanguagesAarakocra",
+      abyssal: "DND5E.LanguagesAbyssal",
+      celestial: "DND5E.LanguagesCelestial",
+      deep: "DND5E.LanguagesDeepSpeech",
+      draconic: "DND5E.LanguagesDraconic",
+      druidic: "DND5E.LanguagesDruidic",
```
Comment (krbz999) [inline module/config.mjs:1716]: I feel like Druidic falls under the same category (that is, no category) as Thieves' Cant.

Comment (Fyorl) [reply module/config.mjs:1716]: I think the PHB does refer to them as 'secret' languages if we wanted to give them a category. But I am ok with leaving them at the top level too. It seems pretty unlikely that you would ever want to gain proficiency with 'all secret languages' given that Thieves' Cant and Druidic are totally unrelated.

Comment (Fyorl) [reply module/config.mjs:1716]: Oh, also, this is marked as resolved but I can still see Druidic under the exotic category in the PR.

Comment (krbz999) [reply module/config.mjs:1716]: > It seems pretty unlikely that you would ever want to gain proficiency with 'all secret languages'

Handy shorthand if one was ever to gain temporary knowledge of 'All languages', they'd only need common, exotic, and secret.

File: module/documents/actor/trait.mjs [hunk 2]
```diff
@@ -383,8 +384,11 @@ export function keyLabel(key, config={}) {
   if ( lastKey === "*" ) {
     let type;
     if ( parts.length ) {
-      const category = CONFIG.DND5E[traitConfig.configKey ?? trait]?.[parts.pop()];
-      if ( !category ) return key;
+      let category = traitData;
+      do {
+        category = (category.children ?? category)[parts.shift()];
+        if ( !category ) return key;
+      } while (parts.length);
```
Comment (Fyorl) [inline module/documents/actor/trait.mjs:391]:
```suggestion
      } while ( parts.length );
```

File: module/documents/actor/trait.mjs [hunk 3]
```diff
@@ -410,9 +414,20 @@ export function keyLabel(key, config={}) {
       if ( index ) return index.name;
       break;
     }
-  }
 
-  return key;
+    // Explicit categories (e.g. languages)
```
Comment (Fyorl) [inline module/documents/actor/trait.mjs:418]: Not clear to me how this differs from the above tool options where we search for an exact category match or a child match? Do we still need those, or are they covered by this new `searchCategory`?

**Review by Fyorl — CHANGES_REQUESTED** [review]

File: module/documents/actor/trait.mjs (hunk 3 above)
Comment (arbron) [reply module/documents/actor/trait.mjs:418]: For vehicle categories in tools, those are stored in separate configuration objects, while these are nested categories in a single object. The joys of years of legacy configuration data :).

Comment (Fyorl) [reply module/documents/actor/trait.mjs:418]: Got it, thanks, that makes sense.

**Review by Fyorl — APPROVED** [review]

**Pull request merged**